### ISSUE TITLE
Slow transfer on many small files - 45% duty cycle

### ISSUE THREAD

**Issue by justinormont** [microsoft/AzureStorageExplorer#2767]
**Storage Explorer Version**: v1.12.0
**Build Number**: 20200116.10
**Platform/OS**: Windows Server 2016
**Architecture**: ia32
**Regression From**:  

## Bug Description ##

Only ~45% of the time is spent sending files.

![image](https://user-images.githubusercontent.com/4080826/77903513-d88b5880-7237-11ea-9f6c-24029025d28d.png)
![image](https://user-images.githubusercontent.com/4080826/77905122-90ba0080-723a-11ea-8f17-fc29f1430497.png)

Network is mostly idle for ~50 seconds, then files are transferred for ~40 seconds in cycles. Note the CPU usage in red shows minimal CPU usage when files are not being transferred.

## Steps to Reproduce ##
~5 million files (200KB on avg, ~1TB total) are being transferred. Files are stored on another server in the same physical server rack. Network is a 1Gb link. When sending, rate is ~500Mbps.

## Expected Experience ##
Network throughput should be fully utilized. 

## Actual Experience ##
Network throughput is pulsed.

## Additional Context ##
In addition, only ~1/2 of the throughput is used, 500Mbps of 1Gbps, when the transfer is occurring, leading to a total efficiency of ~23%.

**Comment by MRayermannMSFT** [comment]
@justinormont is this for blobs, file shares, or something else?

**Comment by justinormont** [comment]
> @justinormont is this for blobs, file shares, or something else?

I am uploading to a blob store.

**Comment by JasonYeMSFT** [comment]
Could you try to record the network requests using tools (e.g. Fiddler) for a problematic transfer? It would be helpful if we know if AzCopy sends any requests during the "idle" periods. Also please share the AzCopy log file with us as well. My rough guess is that AzCopy is waiting on something for ~1min and is only able to proceed after it times out.

**Comment by MRayermannMSFT** [comment]
@justinormont , feel free to reach out to me internally and I can put you in touch with the azcopy team to get help tuning AzCopy for your needs. Alternatively (and for anyone following who isn't internal) you can open issues at their GitHub here to get tuning help as well: https://github.com/Azure/azure-storage-azcopy

**Comment by justinormont** [comment]
Thanks; I pinged you internally. 

Should this issue be transferred to the AzCopy repo? Are there improvement to be made on the Storage Explorer side? For instance, launching multiple copies of AzCopy to work in parallel, or setting more optimal parameters for AzCopy?

 I'd rather not tweak settings and have either Storage Explorer or AzCopy self-optimize.

**Comment by MRayermannMSFT** [comment]
@justinormont can we close this issue?

**Comment by justinormont** [comment]
If the fixes are needed purely in AzCopy, then yes.

**Comment by MRayermannMSFT** [comment]
Ok well, since you've gotten in touch with AzCopy we're going to go ahead and close this. If there's something actionable you have for us after you discuss perf issues with them, feel free to open a new issue describing that action. Thanks.
